src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -210,10 +210,14 @@ async def infer_python_dependencies_via_imports(
 
     unowned_dependency_behavior = python_infer_subsystem.unowned_dependency_behavior
     if unowned_imports and unowned_dependency_behavior is not UnownedDependencyUsage.DoNothing:
+        unowned_imports_with_lines = [
+            f"{modname} ({request.sources_field.address.filename}:{detected_imports[modname]})"
```

Comment:
Right now I think this will be relative to Pants root. Since Pants commands are generally run from Pants root, I think that's fine.

Reply:
This is not safe to use `address.filename` - it will break when you have an explicit `python_source` target. Instead, use `request.sources_field.file_path`, which works because `sources_field` should be `PythonSourceField` which is itself `SingleSourceField`.

Reply:
And yeah, sounds good for path to be relative to build root :)